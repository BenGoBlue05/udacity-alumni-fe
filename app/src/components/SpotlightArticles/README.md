## SpotlightArticles Component
A component that ...

### Example

```js
<SpotlightArticles />
```

### Props

| Prop          | Type     | Default     | Possible Values
| ------------- | -------- | ----------- | ---------------------------------------------
| **myProp**    | String   |             | Any string value


### Other Information
