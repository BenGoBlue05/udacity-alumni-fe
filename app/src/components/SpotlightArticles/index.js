import React, { PropTypes } from 'react';
import styles from './index.module.scss';
import cssModules from 'react-css-modules';
import Heading from 'grommet-udacity/components/Heading';
import Paragraph from 'grommet-udacity/components/Paragraph';
import Box from 'grommet-udacity/components/Box';
import { Link } from 'react-router';

const SpotlightArticles = ({
  articles,
}) => (
  <Box pad="small" direction="column">
    <Heading tag="h3" align="center">
      Spotlight Articles
    </Heading>
    <Box pad={{ vertical: 'small' }} direction="row">
      {articles.map((article, i) =>
        <Box
          key={i}
          align="center"
          basis="1/3"
          pad={{ horizontal: 'small' }}
        >
          <img src={article.feature_image} />
          <Paragraph>
            {`${article.content.slice(0, 200)}...`}
          </Paragraph>
          <span>
            {`Posted by ${article.user.name}`}
          </span>
          <Link to={`/articles/${article.id}`} >Read More</Link>
        </Box>
      )}
    </Box>
  </Box>
);

SpotlightArticles.propTypes = {
  articles: PropTypes.array.isRequired,
};

export default cssModules(SpotlightArticles, styles);
