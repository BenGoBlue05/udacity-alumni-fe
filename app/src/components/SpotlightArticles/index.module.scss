.featuredImage {
  width: 400px;
  height: auto;
  max-width: 100vw;
}

.careerResourcesBlurb {
  max-width: 250px;
}


.featuredArticle {
  @media screen and (min-width: 768px) {
    flex-basis: 330px !important;
  }
  @media screen and (max-width: 44.9375em) {
    flex-direction: row !important;
    flex-basis: 192 !important;
    max-width: 85vw;
    box-sizing: border-box;
  }
}
