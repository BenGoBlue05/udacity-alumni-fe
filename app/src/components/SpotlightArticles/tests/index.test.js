import SpotlightArticles from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<SpotlightArticles />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <SpotlightArticles
        articles={[
          {
            image: 'http://helloworld.com',
            content: 'Hello World',
            updated_at: '12',
            user: {
              name: 'Ryan Collins',
            },
            slug: 'hello-world',
          },
        ]}
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
