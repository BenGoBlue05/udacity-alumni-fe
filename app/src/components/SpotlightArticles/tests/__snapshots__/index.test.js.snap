exports[`<SpotlightArticles /> should render with default props 1`] = `
<Box
  announce={false}
  direction="column"
  focusable={true}
  pad="small"
  responsive={true}
  tag="div">
  <Heading
    align="center"
    tag="h3">
    Spotlight Articles
  </Heading>
  <Box
    announce={false}
    direction="row"
    focusable={true}
    pad={
      Object {
        "vertical": "small"
      }
    }
    responsive={true}
    tag="div">
    <Box
      announce={false}
      basis="1/3"
      direction="column"
      focusable={true}
      pad={
        Object {
          "horizontal": "small"
        }
      }
      responsive={true}
      tag="div">
      <img />
      <Paragraph>
        Hello World...
      </Paragraph>
      <span>
        Posted by Ryan Collins
      </span>
      <Button
        href="/articles/hello-world"
        label="Read More"
        type="button" />
    </Box>
  </Box>
</Box>
`;
